extension/test/test-logger.js

```diff
@@ -27,7 +33,12 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
     }
     finally {
       istream.close();
-      message = JSON.parse(message);
+      try {
+        message = JSON.parse(message);
+      } catch (e) {
+        Cu.reportError("Invalid JSON string: " + message);
+        throw e;
+      }
```

Comment:
`assert.doesntThrow()` or similar should be used there, but as I see, there is only `assert.throws()` exists in SDK.

This piece of code shouldn't throw, because all message should be valid JSON string.